src/tree/render.js

```diff
@@ -102,65 +120,188 @@ const reRenderTree = ({ svg, activeNode, allNodes, o, width, height }) => {
+          secondCurve = `${end.x}  ${start.y + l.y + r.y} ${end.x}  ${start.y + l.y + r.y} ${end.x} ${end.y - l.y}`;
+        } else {
+          l = {x: inverse.x * (size.x / 2 - rAbs) , y: inverse.y * (size.y - rAbs * 2)}
+          firstLine = `${start.x + l.x} ${start.y}`;
+          firstCurve = `${start.x + l.x + r.x} ${start.y} ${start.x + l.x + r.x} ${start.y} ${start.x + l.x + r.x} ${start.y + r.y}`;
+          secondCurve = `${start.x + l.x + r.x} ${end.y} ${start.x + l.x + r.x} ${end.y} ${end.x - l.x} ${end.y}`;
+        }
+
         return (
           `
-          M ${self.x} ${self.y}
-          L ${self.x} ${self.y + h * yrvs}
-          C  ${self.x} ${self.y + h * yrvs + r * yrvs} ${self.x} ${self.y + h * yrvs + r * yrvs} ${self.x + r * xrvs} ${self.y + h * yrvs + r * yrvs}
-          L ${self.x + w * xrvs + r * xrvs} ${self.y + h * yrvs + r * yrvs}
-          C ${parent.x}  ${self.y + h * yrvs + r * yrvs} ${parent.x}  ${self.y + h * yrvs + r * yrvs} ${parent.x} ${parent.y - h * yrvs}
-          L ${parent.x} ${parent.y}
+          M ${start.x} ${start.y}
+          L ${firstLine}
+          C ${firstCurve}
+          L ${start.x + l.x + r.x} ${start.y + l.y + r.y}
+          C ${secondCurve}
+          L ${end.x} ${end.y}
           `
         );
       }
     });
 
   // Zooming and positioning of the tree
-  const bBox = svg._groups[0][0].getBBox();
-  const xFactor = bBox.width / width;
+  const bBox = getBBoxOfNodes(nodes);
+  const scaleToWidhth = bBox.width / width > bBox.height / height;
+  const scaleFactor = scaleToWidhth ? bBox.width / width : bBox.height / height;
+  const translation = scaleToWidhth
+    ? `${-bBox.x} ${-bBox.y + (height * scaleFactor - bBox.height) / 2}`
+    : `${-bBox.x + (width * scaleFactor - bBox.width) / 2} ${-bBox.y}`;
   svg
     .transition()
     .duration(transitionTime)
-    .attr('transform', `scale(${1 / xFactor}) translate(${-bBox.x} ${-bBox.y})`);
+    .attr('transform', `scale(${1 / scaleFactor}) translate(${translation})`);
 };
 
 const renderTree = async ({ element, layout, app, model }) => {
   const b = element.getBoundingClientRect();
   element.innerHTML = '';
   const width = b.width;
   const height = b.height;
-  const center = (width - nodeSize.width) / 2;
 
   // This would allow for different orientations of the tree structure (not needed for now)
-  const orientations = {
-    'top-to-bottom': {
-      size: [width, height],
-      x: function(d) {
-        // return d.x;
-        d.xActual =
-          d.parent && d.parent.xActual
-            ? d.parent.xActual +
-              halfNodeWidth +
-              halfSpaceBetweenNodes +
-              (d.data.childNumber - d.parent.children.length / 2) * (nodeSize.width + spaceBetweenNodes)
-            : center;
-        return d.xActual;
-      },
-      y: function(d) {
-        return d.y;
-      },
-    },
-  };
+  let orientations;
```

Comment:
Not sure this is the best way to do it. I guess you can have multiple options inside the the orientations object, but that requires rewriting a lot of other stuff. @Caele thoughts?